### PR TITLE
Add commonly used libs

### LINKED ISSUE #13: Add commonly used libraries
Add libraries that are commonly used on ustwo projects

- Auto value
- Retrofit
- Stetho
- Leakcanary
- Paperwork
- Timber
- Mockito
- Butterknife
- JSR-310

### PR DESCRIPTION
**Pull request by Numan1617**
Closes #13.

Adds commonly used libs / libs that will help build processes.

### CODE REVIEW

File: dependency.gradle [hunk 1]
```diff
@@ -25,17 +46,44 @@ ext.versions = [
...
 
         dagger: "com.google.dagger:dagger:$versions.dagger",
         daggerCompiler: "com.google.dagger:dagger-compiler:$versions.dagger",
 
         rxJava: "io.reactivex.rxjava2:rxjava:$versions.rxJava",
         rxLint: "nl.littlerobots.rxlint:rxlint:$versions.rxLint",
 
+        autoValue: "com.google.auto.value:auto-value:$versions.autoValue",
+
         // Test dependencies
```
Comment (mshipton) [inline dependency.gradle]: Dependencies need re-ordering a bit, there are app based dependencies in the test area.
Perhaps `App dependencies` would make a better section heading too?

File: dependency.gradle [hunk 2]
```diff
@@ -11,10 +11,31 @@ ext.versions = [
         rxJava: '2.0.0',
         rxLint: '1.0',
 
+        autoValue: '1.2',
```
Comment (mshipton) [inline dependency.gradle:13]: Is there a plan to keep these up to date?

Comment (Numan1617) [reply dependency.gradle:13]: The plan would be any time a project updates to update here too. I can't think of an easy management method but I think that's something we'll work out as we go along.

File: dependency.gradle (hunk 1 above)
Comment (Numan1617) [reply dependency.gradle]: Re-ordered and re-sectioned

**Review by SachseInTheHub — APPROVED** [review]
LGTM

**Pull request merged**